dts/bindings/arm/arm,ethos-u.yaml

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) 2022 Arm Limited.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+description: Arm Ethos-U NPU device tree binding
```

Comment:
Sorry for being vague. When I asked "Could you please explain more here?" in https://github.com/zephyrproject-rtos/zephyr/pull/42580#discussion_r890621441, I more meant information like:

 - what is Ethos-U?
 - links to more information
 - whether users should define their own such nodes when configuring their devicetree, or if they should expect their board to define it for them and just modify an existing definition

Adding "Arm" at the beginning and "device tree binding" at the end is redundant without really clarifying what this binding is for -- does that make sense?

Reply:
I have fleshed out the description a bit, clarifying what Ethos-U is and that it is expected to be configured part of the board device tree file.

I am a bit reluctant to add links to the description, as links tend to change and break over time.

Please have a look in next patch that is pushed.